### PR TITLE
[23.05] dnsmasq: version 2.90

### PR DESCRIPTION
**Pull request by stangri**
Bump to 2.90 to get upstream's fix for DNSSEC KeyTrap (CVE-2023-50387, CVE-2023-50868) among many other goodies and fixes (notably, upstream 568fb024... fixes a UAF in cache_remove_uid that was routinely crashing dnsmasq in my deployment).

Catch up our 200-ubus_dns.patch, too.

Signed-off-by: Nathaniel Wesley Filardo <nwfilardo@gmail.com>
(cherry picked from commit 838a27f64f56e75aae98a3ab2556856224d48d8b)

### CODE REVIEW

**Review by systemcrash — APPROVED** [review]

File: package/network/services/dnsmasq/Makefile [hunk 1]
```diff
@@ -8,13 +8,13 @@
 include $(TOPDIR)/rules.mk
 
 PKG_NAME:=dnsmasq
-PKG_UPSTREAM_VERSION:=2.89
+PKG_UPSTREAM_VERSION:=2.90
 PKG_VERSION:=$(subst test,~~test,$(subst rc,~rc,$(PKG_UPSTREAM_VERSION)))
 PKG_RELEASE:=4
```
[the remark this answers is not in the document]
Comment (stangri) [reply package/network/services/dnsmasq/Makefile:13]: Not in a cherry-pick.

Comment (hardfalcon) [inline package/network/services/dnsmasq/Makefile:13]: `PKG_RELEASE` should also be reset to `1`.

Comment (stangri) [reply package/network/services/dnsmasq/Makefile:13]: Please address your feedback to the original commit this is cherry-picked from.

Comment (systemcrash) [reply package/network/services/dnsmasq/Makefile:13]: This is a cherry pick. Little or nothing should be changed from the base commit. Resolve this comment.

Comment (hardfalcon) [reply package/network/services/dnsmasq/Makefile:13]: Done.

**Comment by systemcrash** [comment]
See also 694e6477848eade21851ec27d90c173b373099fc (@stangri you I think can include this commit in your PR )

**Comment by schuettecarsten** [comment]
Remember to reset PKG_RELEASE to 1 ... ;-) - see  https://github.com/openwrt/openwrt/commit/694e6477848eade21851ec27d90c173b373099fc

File: package/network/services/dnsmasq/Makefile (hunk 1 above)
Comment (schuettecarsten) [inline package/network/services/dnsmasq/Makefile:13]: version bump -> set PKG_RELEASE back to 1

**Comment by stangri** [comment]
> See also [694e647](https://github.com/openwrt/openwrt/commit/694e6477848eade21851ec27d90c173b373099fc) (@stangri you I think can include this commit in your PR )

Thanks @systemcrash I hope I didn't eff it up.

**Review by egc112 — APPROVED** [review]
I agree with @schuettecarsten to set PKG_RELEASE to 1
I had to add a newline at the end.
Otherwise it compiled and ran fine on my EA8500, thanks @stangri

**Review by systemcrash — APPROVED** [review]

**Comment by dave14305** [comment]
For people using dnsmasq-full with dnssec, there is a bug in logging that creates log spam. Fixed upstream by https://thekelleys.org.uk/gitweb/?p=dnsmasq.git;a=commitdiff;h=1ed783b8d7343c42910a61f12a8fc6237eb80417

**Comment by stangri** [comment]
@MkQtS @hardfalcon @schuettecarsten @egc112 -- this is a **cherry-pick** PR, I can include other commits from main branch but I'm not adding anything which was not already merged in main.

@dave14305 thank you, if/when the fix is merged into master, I can add another commit to this PR (if it's not merged by then).

**Comment by systemcrash** [comment]
> For people using dnsmasq-full with dnssec, there is a bug in logging that creates log spam. Fixed upstream by https://thekelleys.org.uk/gitweb/?p=dnsmasq.git;a=commitdiff;h=1ed783b8d7343c42910a61f12a8fc6237eb80417

This should go in as an own commit in master. If that gets cherry picked and supersedes this, that's fine. 

I don't understand why this in itself should take such a long time.

**Review by BKPepe — APPROVED** [review]

**Comment by hauke** [comment]
I created a PR for the extra patches: #14927

**Comment by dave14305** [comment]
> if/when the fix is merged into master, I can add another commit to this PR (if it's not merged by then).



> This should go in as an own commit in master. If that gets cherry picked and supersedes this, that's fine.

Understood, I was just thinking if we introduce log spam to the stable branch, it could create false “panic”, but again, the population of dnssec users using the full variant is probably small.

**Comment by GetVladimir** [comment]
There is a new PR that's a continuation of this update. It allows for the full feature set of the new options from Dnsmasq 2.90 to be configurable via uci on OpenWrt: #14975 

It adds the options for:
--filter-rr
--cache-rr
--dns-rr

and also adds a custom Dnsmasq configuration field via uci that supports any valid Dnsmasq option, so that we don't have to add each feature manually on every update like the ones above, and hopefully makes configurations a little bit easier.

If possible, please review the PR #14975 or add a suggestion if you think there is anything that can be improved or needs to be changed.

The main post for the PR and the details on how to try out and revert back the files is on the OpenWrt Forum: https://forum.openwrt.org/t/add-dnsmasq-custom-options-field-in-luci-gui/193184/18

**Pull request merged**